src/services/getEvmBalances.ts

```diff
@@ -136,11 +142,36 @@ export const getAllEvmTokensBalance = async (
     const tokensNotSupportingMulticall = splittedTokensByMultiCallSupport[0];
     const tokensSupportingMulticall = splittedTokensByMultiCallSupport[1];
 
-    const tokensMulticall = await getTokensBalanceSupportingMultiCall(
-      tokensSupportingMulticall,
-      userAddress as ContractAddress
+    const tokensByChainId = tokensSupportingMulticall.reduce(
+      (groupedTokens, token) => {
+        if (!groupedTokens[token.chainId]) {
+          groupedTokens[token.chainId] = [];
```

Comment:
```suggestion
          groupedTokens[token.chainId] = [token];
```

this can be simplified